logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -21,6 +21,10 @@ def initialize
       # We keep the structured cache to allow
       # the api to search the content of the differents nodes
       @store = Concurrent::Map.new
+
+      # This hash has only one dimension
+      # and allow fast retrieval of the metrics
+      @fast_lookup = Concurrent::Map.new
```

Comment:
does this means we're keeping an internal cache for keys? is this a performance oriented optimization? .... I might not be for this change, but what about having actual pointers and not a copy of the metrics stored in two places? what do you think?